models/DataObject/AbstractObject.php

```diff
@@ -824,6 +824,21 @@ public function save()
             }
             $this->clearDependentCache($additionalTags);
 
+            // if the path changed, refresh the inherited properties before updating dependencies
+            if ($differentOldPath){
+                $this->renewInheritedProperties();
+            }
+            self::updateDependendencies($this);
+
+            // refresh the inherited properties and update dependencies of each children
+            if ($differentOldPath && isset($updatedChildren) && is_array($updatedChildren)) {
+                foreach ($updatedChildren as $updatedDocument) {
```

Comment:
Change the name to $updatedDataObject

Reply:
nice catch, however used shorter `object` suffix